README.md

```diff
@@ -64,6 +64,6 @@ make # or "make server" for packages specific to building the iRODS server.
 # FAQ
 
 ### Q. Can I build an externals package using a different repository?
-Yes. For the externals package of interest, simply set the `"github_repo"` property to the Github URL of repository to clone from. If this property is not present, it means the build will clone the forked repository under https://github.com/irods. The build expects that the clone will have a directory name matching the name of the forked project under https://github.com/irods.
+Yes. For the externals package of interest, simply set the `"git_repository"` property to the URL of the repository to clone. If this property is not present, it means the build will clone the forked repository under https://github.com/irods. The build expects that the clone will have a directory name matching the name of the forked project under https://github.com/irods.
 
-Defining a different Github repository to use for an external normally means a change in the version. For that reason, it is important to remember that `"commitish"` is required to match a branch name or tag. SHA's do not work.
+Defining a different git repository to use for an external normally means a change in the version. For that reason, it is important to remember that `"commitish"` is required to match a **branch name** or **tag**. SHA's do not work.
```

Comment:
> SHA's do not work

is this true?

Reply:
yes. i'm adding a knob that allows a SHA to be passed.

this is due to the use of cloning a specific branch via the --branch option.

Reply:
we need a good name for the new option. the current PR introduces "using_sha" as the name of the option.

the name correctly captures the primary reason for its use, but it isn't limited to one usage pattern. what it really does is separate the fetch/checkout step from the clone step.

suggestions:
- `execute_fetch_checkout_separately`
- `fetch_checkout_separately`
- `using_sha`
- `commitish_is_sha` (this is horrible)
- `clone_separately` (this is horrible too)

Reply:
oooh, i didn't realize `--branch` didn't allow SHAs.   interesting.

Reply:
what if we always just do the clone, then cd, then checkout.     and skip using the `--branch`?

then no additional option needed.

Reply:
sure we can do that. the reason it was added was so we cloned only the things related to that specific commit, which is really only a problem when handling large repos with a lot of git history.

Reply:
do we want to go back to the old behavior?

Reply:
Ah, I see.    That does seem worth it...

How about...

`use_specific_sha`
or just
`use_sha`

Reply:
i like it

Reply:
renamed to `enable_sha`.